docs/data/data-grid/state/RestoreStateApiRef.tsx

```diff
@@ -0,0 +1,339 @@
+        <form onSubmit={handleSubmitForm}>
+          <DialogTitle>New custom view</DialogTitle>
+          <DialogContent>
+            <TextField
+              autoFocus
+              value={label}
+              onChange={onLabelChange}
+              margin="dense"
+              size="small"
+              label="Custom view label"
+              variant="standard"
+              fullWidth
+            />
+          </DialogContent>
+          <DialogActions>
+            <Button type="button" onClick={() => setIsAddingView(false)}>
+              Cancel
+            </Button>
+            <Button type="submit" disabled={!isValid}>
+              Create view
+            </Button>
+          </DialogActions>
+        </form>
+      </Dialog>
+    </React.Fragment>
+  );
+};
+
+const CustomToolbar = () => {
+  const apiRef = useGridApiContext();
+  const [state, dispatch] = React.useReducer(demoReducer, DEMO_INITIAL_STATE);
+
+  const createNewView = () => {
+    dispatch({
+      type: 'createView',
+      value: apiRef.current.exportState(),
+    });
+  };
+
+  const handleNewViewLabelChange = (
+    e: React.ChangeEvent<HTMLInputElement | HTMLTextAreaElement>,
+  ) => {
+    dispatch({ type: 'setNewViewLabel', label: e.target.value });
+  };
+
+  const handleDeleteView = React.useCallback((viewId: string) => {
+    dispatch({ type: 'deleteView', id: viewId });
+  }, []);
+
+  const handleSetActiveView = (viewId: string) => {
+    apiRef.current.restoreState(state.views[viewId].value);
+    dispatch({ type: 'setActiveView', id: viewId });
+  };
+
+  const handlePopperAnchorClick = (event: React.MouseEvent) => {
+    dispatch({ type: 'togglePopper', element: event.currentTarget as HTMLElement });
+    event.stopPropagation();
+  };
+
+  const handleClosePopper = () => {
+    dispatch({ type: 'closePopper' });
+  };
+
+  const isNewViewLabelValid = React.useMemo(() => {
+    if (state.newViewLabel.length === 0) {
+      return false;
+    }
+
+    return Object.values(state.views).every(
+      (view) => view.label !== state.newViewLabel,
+    );
+  }, [state.views, state.newViewLabel]);
+
+  const canBeMenuOpened = state.isMenuOpened && Boolean(state.menuAnchorEl);
+  const popperId = canBeMenuOpened ? 'transition-popper' : undefined;
+
+  return (
+    <GridToolbarContainer style={{ height: 48 }}>
+      <Button
+        aria-describedby={popperId}
+        type="button"
+        size="small"
+        onClick={handlePopperAnchorClick}
+      >
+        Custom view ({Object.keys(state.views).length})
+      </Button>
+      <ClickAwayListener onClickAway={handleClosePopper}>
+        <Popper
+          id={popperId}
+          open={state.isMenuOpened}
+          anchorEl={state.menuAnchorEl}
+          transition
+          placement="bottom-start"
+        >
+          {({ TransitionProps }) => (
+            <Fade {...TransitionProps} timeout={350}>
+              <Paper>
+                <List>
```

Comment:
`List` is not accessible. When I open it I can't navigate through the items via keyboard. [`Menu`](https://mui.com/components/menus/#selected-menu) handles that. [`MenuList`](https://mui.com/components/menus/#menulist-composition) can also be used. We use the last to be able to still use a `Popper`

Reply:
I migrated to a `MenuList`